packages/nx/src/command-line/init/configure-plugins.ts

```diff
@@ -0,0 +1,210 @@
+import * as createSpinner from 'ora';
+import { bold } from 'chalk';
+
+import {
+  getPackageManagerCommand,
+  PackageManagerCommands,
+} from '../../utils/package-manager';
+import { GitRepository } from '../../utils/git-utils';
+import { output } from '../../utils/output';
+import { flushChanges, FsTree } from '../../generators/tree';
+import {
+  Generator as NxGenerator,
+  GeneratorCallback,
+  GeneratorsJsonEntry,
+} from '../../config/misc-interfaces';
+import { getGeneratorInformation } from '../generate/generator-utils';
+import { workspaceRoot } from '../../utils/workspace-root';
+import { addDepsToPackageJson, runInstall } from './implementation/utils';
+import { getPluginCapabilities } from '../../utils/plugins';
+import { isAngularPluginInstalled } from '../../adapter/angular-json';
+
+export function runPackageManagerInstallPlugins(
+  repoRoot: string,
+  pmc: PackageManagerCommands = getPackageManagerCommand(),
+  plugins: string[]
+) {
+  if (plugins.length === 0) {
+    return;
+  }
+  addDepsToPackageJson(repoRoot, plugins);
+  runInstall(repoRoot, pmc);
+}
+
+/**
+ * Installs a plugin by running its init generator. It will change the file system tree passed in.
+ * @param plugin The name of the plugin to install
+ * @param repoRoot repo root
+ * @param verbose verbose
+ * @param options options passed to init generator
+ * @returns void
+ */
+export async function installPlugin(
+  plugin: string,
+  repoRoot: string = workspaceRoot,
+  verbose: boolean = false,
+  options: { [k: string]: any }
+): Promise<void> {
+  const host = new FsTree(repoRoot, verbose, `install ${plugin}`);
+  const capabilities = await getPluginCapabilities(repoRoot, plugin, {});
+  const generators = capabilities?.generators;
+  if (!generators) {
+    throw new Error(`No generators found in ${plugin}.`);
+  }
+
+  const initGenerator = findInitGenerator(generators);
+  if (!initGenerator) {
+    output.log({
+      title: `No "init" generator found in ${plugin}. Skipping initialization.`,
+    });
+    return;
+  }
+  const { implementationFactory } = getGeneratorInformation(
+    plugin,
+    initGenerator,
+    repoRoot,
+    {}
+  );
+
+  const implementation: NxGenerator = implementationFactory();
+  const task: GeneratorCallback | void = await implementation(host, options);
+  if (task) {
+    await task();
+  }
+  flushChanges(repoRoot, host.listChanges());
```

Comment:
This should be the other way around.

Flush then run tasks.

Reply:
changed